src/SprykerEco/Yves/AmazonPay/Theme/default/components/atoms/amazon-order-submitter/amazon-order-submitter.ts

```diff
@@ -0,0 +1,92 @@
+import Component from 'ShopUi/models/component';
+
+declare const window: {
+    OffAmazonPayments: {
+        initConfirmationFlow(sellerId: string, orderId: string, callback: (confirmationFlow: ConfirmationFlow) => void): void
+    }
+};
+
+interface ConfirmationFlow {
+    success(): void;
+    error(): void;
+}
+
+const XHR_SUCCESS_CODE = 200;
+const XHR_REDIRECT_CODE = 302;
+
+export default class AmazonOrderSubmitter extends Component {
+    protected button: HTMLButtonElement;
+    protected xhr: XMLHttpRequest;
+
+    constructor() {
+        super();
+        this.xhr = new XMLHttpRequest();
+    }
+
+    protected readyCallback(): void {}
+
+    protected init(): void {
+        this.button = <HTMLButtonElement>this.getElementsByClassName(`${this.jsName}__button`)[0];
+        this.mapEvents();
+    }
+
+    protected mapEvents(): void {
+        this.button.addEventListener('click', this.onButtonClick.bind(this));
+    }
+
+    protected placeOrder<T = string>(confirmationFlow: ConfirmationFlow): void {
+        new Promise<T>((resolve, reject) => {
```

Comment:
Please add types for `resolve`, `reject`. Check it everywhere.

Reply:
fixed